dotnet/src/Extensions/PromptTemplate.Handlebars/Helpers/KernelHelperUtils.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+
+namespace Microsoft.SemanticKernel.PromptTemplates.Handlebars.Helpers;
+
+/// <summary>
+/// Extension class to register additional helpers as Kernel System helpers.
+/// </summary>
+internal static class KernelHelpersUtils
+{
+    /// <summary>
+    /// Determines whether the specified type is a numeric type.
+    /// </summary>
+    /// <param name="type">The type to check.</param>
+    /// <returns>True if the type is a numeric type; otherwise, false.</returns>
+    public static bool IsNumericType(Type? type)
+    {
+        return type is not null &&
+                Type.GetTypeCode(type) is
+                    TypeCode.SByte or
+                    TypeCode.Int16 or
+                    TypeCode.Int32 or
+                    TypeCode.Int64 or
+                    TypeCode.Byte or
+                    TypeCode.UInt16 or
+                    TypeCode.UInt32 or
+                    TypeCode.UInt64 or
+                    TypeCode.Double or
+                    TypeCode.Single or
+                    TypeCode.Decimal;
```

Comment:
This should probably also include `type is nint || type is nuint`

Reply:
Seeing this warning when I add this:
`[CS8121](classifiedtextelementview-invokeaction:0): An expression of type 'Type' cannot be handled by a pattern of type 'nint'.`

![Uploading Screenshot 2023-12-11 at 8.39.01 AM.png…]()

Reply:
Sorry, I was typing too quickly, it should be:
```C#
return
    type == typeof(nuint) ||
    type == typeof(nint) ||
    ... // what you currently have
```